src/components/Switch/index.js

```diff
@@ -11,23 +11,26 @@ class Switch extends React.Component {
         };
     }
     handleClick(which) {
-        this.setState({chosenParamIndex: which })
+        this.setState({
+            chosenParamIndex: which,
+        }); 
     }
 
     getClass(index) {
-        return "Switch__Param " + (this.state.chosenParamIndex === index ? "Switch_activeParam" : "");
+        const activeParamClass = this.state.chosenParamIndex === index ? "switch__param--active" : "";
+        return `switch__param ${activeParamClass}`;
     }
 
     render() {
         const { params, text } = this.props;
         return (
-            <div className="Switch">
-                {text}
-                <span className={this.getClass(0)} onClick={() => this.handleClick(0)}>
-                    {params[0]}
+            <div className="switch">
+                { text }
+                <span className={ this.getClass(0) } onClick={ this.handleClick.bind(this, 0) }>
```

Comment:
you misunderstood me :-) this.handleClick.bind(this, 0) does the same thing as arrow function... but I have not noticed before that you needed to pass extra param to handleClick so bind in the constructor doesn't work well in your case... there are several solutions that should work better for similar situations, e.g. you can set index as data attribute to the element and read that index in common handleClick handle or you can extract this element into small component with at least 2 properties - index/number and onClick/func so this component will handle click on the element and pass index to onClick callback so parent component can understand what element handled this click.